src/ruby/ext/grpc/rb_channel_credentials.c

```diff
@@ -138,9 +138,13 @@ static ID id_pem_cert_chain;
     ...
     creds3 = Credentials.new(pem_root_certs, pem_private_key,
                              pem_cert_chain)
+    ...
+    creds4 = Credentials.new(fallback_creds)
```

Comment:
I'm a little worried this API may be confusing to reason about, because we are deciding to create an XDS credentials object or SSL credentials object, based on whether or not the first parameter to this constructor is a credentials object, and returning an object of the exact same type in both cases.

Also, this seems different from [Python](https://github.com/grpc/grpc/pull/25365) and [C++](https://github.com/grpc/grpc/blob/c8c80fd9988fb5e508ae2f6b39e7901c2054d4f6/src/core/lib/security/credentials/xds/xds_credentials.h#L35), which both have `XdsCredentials` and `XdsServerCredentials` as explicit types.

Instead of this API, what do you think about creating a new class, `XdsChannelCredentials`, which takes a non-optional fallback creds parameter. Meanwhile, this constructor here can be left unchanged.

Same comment for the server credentials API and changes made to `ServerCredentials`.

cc @sanjaypujare and @stanley-cheung

Reply:
Is it OK to add a `newXds` function to ChannelCredentials for xds? Usage will be looks like GRPC::Core::ChannelCredentials.newXds(fallback)

Reply:
XdsChannelCredentials class and XdsServerCredentials class are added.
@apolcyn @stanley-cheung